lib/rails_param/param.rb

```diff
@@ -73,7 +73,6 @@ def recurse(params, index = nil)
 
     def coerce(param, type, options = {})
       begin
-        return nil if param.nil?
```

Comment:
Removing this line was the key to solving the bug. However, some test cases were expecting the `coerce` method to return nil. I'm kind of worried about some of the Coerce classes not handling a nil value well but just not having the test coverage for that at this point. `BigDecimal` for example is just going to blow up.

Any ideas on an approach for handling a nil param value better?

Reply:
Short term solution could be to add something like:

```ruby
return nil if param.nil? && type != Array
```

I expect the longer term solution would be to move any type-dependent logic into the relevant coercion classes (and add test cases).

Taking a step back, I wonder if an array specified as `nil` should raise or be coerced to an empty array :thinking: IMO a `null` array is an invalid parameter error but I'm not sure that's the way other people will see it. Thoughts?

Reply:
Yeah that was my first approach but ultimately I want to get all of those early returns out of the `coerce` method.

I'll revert these changes just to the single line now to solve the bug w/ minimal changes.

Longer term: I tend to agree that a nil array seems like an invalid parameter. But it seems less clear that a nil string should be an invalid parameter rather than coerced to an empty string. 🤔

Reply:
Oooh yep that's a good point. Need to think about this a bit more.